lektor/imagetools.py

```diff
@@ -595,33 +584,35 @@ def make_image_thumbnail(
     if format == "svg":
         return Thumbnail(source_url_path, width, height)
 
-    if not upscale:
+    if mode == ThumbnailMode.FIT:
+        computed_width, computed_height = compute_dimensions(
+            width, height, source_width, source_height
+        )
+    else:
+        computed_width, computed_height = width, height
 
-        def _original():
-            return Thumbnail(source_url_path, source_width, source_height)
+    upscaling_requested = (
```

Comment:
Well, the user did request upscaling, in that s/he requested a "thumbnail" larger than the original image.
I struggled a bit with the name, too.  `requires_upscaling` doesn't seem quite right either, since, depending on the setting of `upscale`, we might not actually be performing the upscale — in which case, how "required" (or needed) was it?  (In that case, the user asked for (requested) an upscaled image, but we're giving him back the original because upscaling was forbidden due to the setting of `upscale`.  It would all be clearer if `upscale` were named `allow_upscaling` or similar, but it's probably too late to change that.)

But if you'd like it changed, that's fine.

Reply:
I commented about it because I initially confused it with the input argument `upscale=True`. Maybe something like overly descriptive like `src_smaller_than_thumbnail` then?

Reply:
How about `would_upscale`?

Reply:
Yeah, that works for me!